src/ut/httpresolver_test.cpp

```diff
@@ -154,6 +162,38 @@ TEST_F(HttpResolverTest, AAAARecordResolution)
             HttpRT(_httpresolver).set_host("cpp-common-test.cw-ngv.com").set_port(8888).resolve());
 }
 
+TEST_F(HttpResolverTest, BlacklistTimeLowerBound)
+{
+  add_white_records(11);
+  _httpresolver.blacklist(ip_to_addr_info("3.0.0.0"));
+  cwtest_advance_time_ms(29999);
+  EXPECT_TRUE(is_black("3.0.0.0", 11, 15));
+}
+
+TEST_F(HttpResolverTest, BlackListTimeUpperBound)
+{
+  add_white_records(11);
+  _httpresolver.blacklist(ip_to_addr_info("3.0.0.0"));
+  cwtest_advance_time_ms(30001);
+  EXPECT_TRUE(is_gray("3.0.0.0", 11, 15));
+}
+
+TEST_F(HttpResolverTest, GrayListTimeLowerBound)
+{
+  add_white_records(11);
+  _httpresolver.blacklist(ip_to_addr_info("3.0.0.0"));
+  cwtest_advance_time_ms(59999);
+  EXPECT_TRUE(is_gray("3.0.0.0", 11, 15));
+}
+
+TEST_F(HttpResolverTest, GrayListTimeUpperBound)
+{
+  add_white_records(11);
+  _httpresolver.blacklist(ip_to_addr_info("3.0.0.0"));
+  cwtest_advance_time_ms(60001);
+  EXPECT_TRUE(is_white("3.0.0.0", 11, 15));
+}
```

Comment:
The key point with these tests is the HttpResolver uses a blacklist time of 30s and a graylist time of 30s. That isn't clear from the test case names so you could do with a comment.